GridRNN cell uses tuples for output and states

phvu opened the pull request:
In response to this (https://github.com/tensorflow/tensorflow/issues/2560) and recent changes in LSTMCell, this PR added `state_is_tuple=True` and `output_is_tuple=True` into the constructor of GridRNNCell:

```
> cell = tf.contrib.grid_rnn.Grid2LSTMCell(3, use_peepholes=True)
> cell.state_size
       (LSTMStateTuple(c=3, h=3), LSTMStateTuple(c=3, h=3))
> cell.output_size
       (3,)
```

This means there are 2 LSTM cells in the grid, and the state of each cell has size of `(c=3, h=3)`. There is only one output dimension, whose size is 3.

In contrast:

```
> cell = tf.contrib.grid_rnn.Grid2BasicRNNCell(3)
> cell.state_size
     (3, 3)
> cell.output_size
     (3,)
```

This means there are 2 BasicRNN cell in the grid, and the state of each cell has size of 3. There is only one output dimension, whose size is 3.

Previous behaviour is maintained by using `state_is_tuple=False, output_is_tuple=False` when creating the cell.

```
> cell = tf.contrib.grid_rnn.Grid2LSTMCell(3, use_peepholes=True, state_is_tuple=False, output_is_tuple=False)
    WARNING:tensorflow:<tensorflow.contrib.grid_rnn.python.ops.grid_rnn_cell.Grid2LSTMCell object at 0x10de68f50>: Using a concatenated state is slower and will soon be deprecated.  Use state_is_tuple=True.
    WARNING:tensorflow:<tensorflow.contrib.grid_rnn.python.ops.grid_rnn_cell.Grid2LSTMCell** object at 0x10de68f50>: Using a concatenated output is slower and will soon be deprecated.  Use output_is_tuple=True.

> cell.state_size
    12
> cell.output_size
    3

> cell = tf.contrib.grid_rnn.Grid2BasicRNNCell(3, state_is_tuple=False, output_is_tuple=False)
    WARNING:tensorflow:<tensorflow.contrib.grid_rnn.python.ops.grid_rnn_cell.Grid2BasicRNNCell object at 0x10de4b950>: Using a concatenated state is slower and will soon be deprecated.  Use state_is_tuple=True.
    WARNING:tensorflow:<tensorflow.contrib.grid_rnn.python.ops.grid_rnn_cell.Grid2BasicRNNCell object at 0x10de4b950>: Using a concatenated output is slower and will soon be deprecated.  Use output_is_tuple=True.

> cell.state_size
   6
> cell.output_size
   3
```

This also fixes https://github.com/tensorflow/tensorflow/issues/4296

benoitsteiner committed:
Pull the latest Eigen version that supports OpenCL

phoaivu committed:
removed input_size parameter and fix #4296

phoaivu committed:
removed input_size parameter and fix #4296

phoaivu committed:
removed input_size parameter and fix #4296

phoaivu committed:
added state_is_tuple and output_is_tuple, 5 tests passed, 14 to go

phoaivu committed:
added state_is_tuple and output_is_tuple, 5 tests passed, 14 to go

phoaivu committed:
all tests passed

phoaivu committed:
all tests passed

tensorflow-jenkins commented:
Can one of the admins verify this patch?

StefOe commented on tensorflow/contrib/grid_rnn/python/ops/grid_rnn_cell.py:
```diff
@@ -132,74 +171,118 @@ def __call__(self, inputs, state, scope=None):
+    """
+    conf = self._config
+
+    if inputs is not None and inputs.get_shape().with_rank(2)[1].value > 0 \
```
This indeed fixes #4296

ebrevdo commented:
Jenkins, test this please.

benoitsteiner committed:
Switch to the latest version of Eigen that supports OpenCL

benoitsteiner committed:
Upgraded to the latest version of Eigen

zheng-xq committed:
Merge changes from github.
Change: 137532946

benoitsteiner committed:
Fixed a typo

vrv commented:
@tensorflow-jenkins test this please

vrv commented:
@ebrevdo do you give an LGTM?

tensorflower-gardener committed:
Switch FakeQuant and Dequantize ops in array_ops to use C++ shape functions.
Change: 137756007

tensorflower-gardener committed:
Split code to compute control flow structure of a graph out of graph_partition.cc and into its own module (control_flow.{cc,h}).
Change: 137756517

tensorflower-gardener committed:
The concat error message does not provide enough information. This just changes the error message to include shapes to aid in debugging. Since the shape Tensors should in general be small.
Change: 137756593

langmore committed:
ScaleAndShift Bijector updates/fixes:
DOC:  Clarify that (for the ScaleAndShift bijector) scale * X is matrix
multiplication, and that scale must be triangular and non-singular.
CODE:  Check that the conditions on scale are met.
Change: 137761917

Drag0 committed:
Fixed failed building of /pi_examples/label_image. (#5204)

Included <stdio.h> before including <jpeglib.h> because jpeglib.h implicitly requires it.
Cast 'usage' variable to 'const char*' because tensorflow::port::InitMain() function requires C-style string instead of std::string.

Commit:
Remove unneeded PLATFORM macro usage.
Change: 137762301

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137766763

Commit:
Rename the TensorBoard "events" dashboard to "scalars".

This is a much more sensible name, as the scalar dashboard shows all the scalar data collected by tf.summary.scalar.
Change: 137766817

sherrym committed:
Added clear_devices option for export_scoped_meta_graph().
Change: 137767898

benoitsteiner committed:
Added the ability to wait for queues to start running before returning from
QueueRunner::Start(). This provides a reliable way to check the value of the status_ variable.
Change: 137769682

Commit:
Merge commit for internal changes

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137771513

tensorflower-gardener committed:
Automated rollback of change 137715407
Change: 137771699

sukritiramesh committed:
Add support for clear devices in SavedModel.
Change: 137772478

Commit:
Merge pull request #5313 from vrv/branch_137767898

Branch 137767898

rinugun committed:
Also retry failed Close() in the WritableFile's destructor.
Change: 137774256

Commit:
Make the DimensionHandle version of computing windowed output size
public.
Change: 137774322

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137775024

Commit:
It seems extremely unlikely that the fragment being shaded lies /outside/ of
the point sprite to which it belongs. Because of that, delete a
point-on-side-of-line-segment test and simplify the analytial triangle
renderer.

Additionally, fix the "vampire teeth" effect when rendering the letters M, W, V in canvas labels by setting a very conservative miter limit.
Change: 137776328

gunan committed:
Increase the size of retrying_file_system_test, as it flakily times out.
Change: 137788506

gunan committed:
Disable flaky test_save_secs_saves_periodically.
Change: 137788894

tensorflower-gardener committed:
Increasing the limit of 20 colors for the Embedding Projector to 50.
Some colors are duplicated if there are more than 20.
Change: 137820753

Commit:
Fix picking bug in fragment shader. Since we're not doing a camera-relative
depth sort, we can't do 1-bit alpha correctly. Revert to using discard in the
picking shader until we do a global camera-space z sort (if ever).

Aggregate [PointAccessor, PointAccessor, PointAccessor] as a new type,
PointAccessors3D.
Change: 137823236

ptrendx committed:
Changed context creation to use primary context (#5241)

* Changed context creation to use primary context

* Adding CUDA version checks as per PR comments

benoitsteiner committed:
Merge remote-tracking branch 'upstream/master'

tensorflower-gardener committed:
Don't split nodes in TensorForest until they have accumulated more than 1 class.
Change: 137833057

Commit:
Increase initial zoom in.
Change: 137834159

Commit:
Feedback from #5267 applied.

tensorflower-gardener committed:
Allow passing constant_value_as_shape from call_cpp_shape_fn to the C++ shape
inference. Delegate to C++ shape function for Reshape.

Fix reshape to handle attempting to infer unknown dim when product of known
elements is 0.
Change: 137837591

benoitsteiner committed:
Fixed merge conflicts.

benoitsteiner committed:
Fixed merge conflict

Commit:
Seal tf.python_io's interface.
Change: 137843740

tensorflower-gardener committed:
Load convolution node definitions to transfer graphs to SOC
Change: 137845970

Commit:
Rename tab "projections" to "embeddings" in TensorBoard
Change: 137847545

tensorflower-gardener committed:
Refactor DNNLinearCombinedClassifier from inheritance to composition.
Change: 137848991

Commit:
Seal gfile.py's interface.
Change: 137849082

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137849675

Commit:
Move deprecation module into core framework from contrib/framework
Change: 137850119

tensorflower-gardener committed:
Add better error messaging for passing a non-SparseTensor arg to
SparseTensor.from_value.
Change: 137850511

tensorflower-gardener committed:
Speed up SwapDimension1And2InTensor3UsingTiles.

For all but the edge tiles, we can unroll the loops that read and write
from our shared memory tile.
Change: 137853968

Commit:
Fixes #4226: Allow to specify partial shape in sparse_placeholder.
Change: 137854718

tensorflower-gardener committed:
Store ksize attribute for graph transfer to SOC
Change: 137855838

Commit:
cmake support to run python and cc unittests.

rinugun committed:
Reduce the runtime for the unittest from 50 to 2 seconds.
Change: 137861834

tensorflower-gardener committed:
Basic slow versions of resource-based variable ops which have the right semantics.
Change: 137863207

Commit:
Add fill_lower_triangular utility function.
Change: 137863303

Commit:
Add C++ Coordinator
Change: 137866409

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137866950

tensorflower-gardener committed:
Change quantize_graph in eightbit mode to remove FakeQuantWithMinMaxVars
nodes and use the information provided by them to set the min/max values
on quantization-related nodes.

In eightbit mode, also changed how constant weights are quantized - instead of
doing it as a step after the main recursion, do it during the main recursion.
This allows the float inputs to FakeQuantWithMinMaxVars to be excluded from
quantization.

In eightbit mode, maintain more state in the stack during recursion.

Also change quantize reshape registration to register always and not use
TF_CALL_xyz; this matches other quantized ops.
Change: 137877226

Commit:
Update polymer dependencies.
Change: 137877854

Commit:
Remove the ckpt existence test in test_tutorials.sh.

It's a weak test + causes complications if we consider both V1 & V2.
Change: 137878590

rinugun committed:
Make GetChildren never return a self directory marker.

In GCS, a folder 'gs://bucket/path' can contain a zero-length object 'gs://bucket/path/', which primarily exists to make IsDirectory('gs://bucket/path') return 'true'.

Naturally GetChildren returned the directory marker for the specified directory whenever it existed, represented as an empty string, which violates assumptions that people normally have from this functions.
Change: 137878839

gunan committed:
Fix race conditions in tests due to using same folders for saver.
Change: 137881303

gunan committed:
Disable failing tests from jenkins.
Change: 137883577

Commit:
Fix issue where we wouldn't descend into subdirectories when listing GCS runs
in TensorBoard.

- Fixes GCS so that it still includes prefixes even if items is empty.
- Change list_directory to use GetChildren() instead of GetMatchingFiles.
  GetMatchingFiles in GCS has a quirk where, unlike other filesystems, it
  doesn't include directories. There's no obvious fix given the GCS API.
Change: 137885363

tensorflower-gardener committed:
Remove weight_parameters from OpStats and graph_metrics.
Change: 137885496

tensorflower-gardener committed:
Added a button below each image in the Tensorboard image panel. Clicking this new button shows an actual size version of the image.
Change: 137886562

ebrevdo committed:
Move SparseTensor into its own framework.py, add get_shape().
Change: 137887950

tensorflower-gardener committed:
Add a "libs" attribute to the tf_kernel_libraries BUILD rule.  This is used to
specify additional tf_kernel_library targets that should be grouped into the
omnibus cc_library target that includes all generated rules, but should not be
considered dependencies of each individual generated tf_kernel_library target.
Use the new attribute to split out some existing deps entries and avoid
unnecessary transitive dependencies.

Split out the :l2loss_op library so that it doesn't share the same deps list
with the rest of the "nn" libraries, in particular avoiding a dependency on
:pooling_ops which in turn depends on the expensive :eigen_helpers.
Likewise, split out :lookup_table_{init_}op from the "data_flow" libraries.
Change: 137887998

gunan committed:
Remove references to gcudacc.
Change: 137888607

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137889948

Commit:
Merge commit for internal changes

rinugun committed:
Make GetMachingPaths also return directory names.

This is now consistent with what other file systems have.
Change: 137892031

benoitsteiner committed:
Merge pull request #5339 from vrv/branch_137889948

Branch 137889948

Commit:
TensorBoard: Don't show plot tooltips for invisible lines

Specifically, suppose you have a line chart with 30 runs. If you mouse over, you will have 30 tooltips, which is hard to read. Previously, if you zoom in so that only 1 run is visible on the screen, you would still get 30 tooltips. Now, if you mouse over you will only see tooltips for runs that are actually on the screen.
Change: 137905115

Commit:
Enable the embeddings tab in TensorBoard and release new version.
Change: 137905523

jart committed:
TensorBoard: Ignore deleted files on Colossus
Change: 137907844

caisq committed:
Fix a missing sparse column in wide_and_deep example code
Change: 137908539

nsuke committed:
Consistently use bytes for git version tracking for Python3 compatibility (#5318)

gunan committed:
Fix sourceforge links for TF external dependencies.

gunan committed:
Fix: disable memory sanitizer for quantized_matmul_op_for_hexagon_test.
Change: 137914787

gunan committed:
Fix: Replace sourceforge URLs with debian ftp urls.
Change: 137917462

tensorflower-gardener committed:
Change for internal compatibility.

timharley committed:
Register QueueRunner threads with coordinator at creation time not start time.

This fixes a race condition in coordinator.join(). Removed workaround in
QueueRunnerTest.
Change: 137933187

caisq committed:
Merge pull request #5346 from gunan/master

Fix sourceforge links for TF external dependencies.

Commit:
Clean up the Bookmarks UI:

-Update the help text to not use "bookmarks" as a noun and a verb
-Change the download icon to the save icon
-Disable the save icon if there are no bookmarks
-Change visible nomenclature to save and load instead of upload/download
-Fix the radio button edge-click bug that selects the first bookmark
-Do some basic checks on the shape of the dataset and the bookmarks.
Change: 137943270

hawkinsp committed:
Perform constant folding in the calling thread, rather than using a singleton threadpool. Avoids a potential deadlock when constant folding is called in a re-entrant manner.
Change: 137945911

benoitsteiner committed:
Improved the formatting of the BUILD file

tensorflower-gardener committed:
Use "Example:", not "For example:" in docstrings to introduce an example.
Change: 137952318

benoitsteiner committed:
Deleted unecessary code

Commit:
Add support for user-generated tsv tensor files in the embedding projector.
Change: 137954868

tensorflower-gardener committed:
Change default dominate_method to bootsrap.
Change: 137959663

Commit:
Set initial t-sne perplexity based on dataset size.

Set it to P = sqrt(N) / 4, where N is the number of data points.
Change: 137961121

Commit:
Merge commit for internal changes

burness committed:
add type info described in issue #5236 (#5278)

* add type info when assign

* add INFO

* add spaces

* wrap to 80 lines

tensorflower-gardener committed:
Change v2 session bundle compression method to match v1.
Change: 137961557

tensorflower-gardener committed:
Delete old Eigen threadpool and associated code.
Get rid of EIGEN_USE_NONBLOCKING_THREAD_POOL symbol.
Change: 137961673

tensorflower-gardener committed:
Docstring example and formatting updates
Change: 137961957

benoitsteiner committed:
Placement of operations of SYCL devices needs to be done by using the
/device:sycl string.

tensorflower-gardener committed:
Initialize resources in the prediction path before loading from a checkpoint.
Reload all saveable objects from the graph.
Change: 137964298

Commit:
Make the demo datasets provider fetch a projector config JSON file.

This makes "Smart Reply 5K" the first in the list of tensors, and we now choose the first
tensor in the embeddings array as the default tensor.
Change: 137964537

benoitsteiner committed:
Simplified the addition of devices.
Change: 137964954

tillahoffmann committed:
Hessian with respect to one-dimensional tensors (#5329)

* Add hessian op.

* Add test.

* Fix typos.

* Fix tests.

* Revert "Fix tests."

This reverts commit 8ef3abfc40a312c301bc25ceba2fdc90a5527eb0.

* Fix tests.

* Support multiple tensors in xs.

* Fix docstring formatting.

* Add hessians to training docs.

* Remove gpu loop.

* Add test and bugfix.

* Add name scope.

* Update error messages.

benoitsteiner committed:
Upgraded to the latest version of Eigen
Change: 137969509

Commit:
Merge pull request #5358 from vrv/branch_137959663

Branch 137959663

Drag0 committed:
Changed reference from eigen-latest/ to eigen/ in order to fix the following error: ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:42: fatal error: unsupported/Eigen/CXX11/Tensor: No such file or directory (#5307)

shoyer committed:
Add initial version of tf.contrib.integrate

Includes tf.contrib.integrate.odeint, an integrator for ODEs patterned off of
scipy.integrate.odeint.
Change: 137973526

Commit:
Support automatic differentiation in fill_lower_triangular when input has no batch dims.
Change: 137973576

tensorflower-gardener committed:
Remove ':' from summary names in optimize_loss.
Change: 137974467

Commit:
Merge branch 'master' into cmaketests

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137988710

Commit:
Autogenerated Change: Change TensorBoard TAG to 35
Change: 137989251

tensorflower-gardener committed:
Replace usages of tf.select with tf.where.
Change: 137989913

Commit:
Distributed partial run support.

This change completes support for distributed partial run.
RPC/Proto changes:
- PartialRunSetup RPC is added to the Master interface.
- A is_partial bool is added to the RunStepRequest.

MasterSession implementation:
- When the MasterSession gets a PartialRunSetupRequest, it constructs the graph and registers the partitions.
- When a is_partial RunStepRequest is received, the feeds and fetches are validated, and  corresponding is_partial RunGraphRequests are sent to the workers.
Change: 137989990

Commit:
Make perplexity be at least 5.
Change: 137990808

Commit:
exclude some tests that are failing or not compiling yet.

aselle committed:
Clarify when you need to use Makefile

Fixes #5360
Change: 137993461

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 137995278

itsmeolivia committed:
changing the order of tf.svd's arguments to be the same as numpy's
Change: 137996377

Commit:
Seal *_ops interfaces.
Change: 138001077

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138001884

tensorflower-gardener committed:
Allowed images in the Images panel to approach being pixelated after being scaled in Firefox. This solution is a brief CSS change.
Change: 138005373

gunan committed:
Fix: to avoid flakes, make sure GenerateTestData picks a new folder every time.
Change: 138008252

Commit:
document tensorflow_BUILD_CC_TESTS and tensorflow_BUILD_PYTHON_TEST

sukritiramesh committed:
Update comment related to assets in SavedModel.
Change: 138010189

Commit:
Move the last skflow examples into learn examples. Remove skflow examples folder. Fixes #2606. Deleted usage of learn.models.logistic_regression, learn.ops.*.
Change: 138010275

tillahoffmann committed:
Add log1p (#5356)

* Add log1p.

* Use log1p in cross entropy.

* Register log1p in math_ops.

* Update copyright.

* Fix tests.

tensorflower-gardener committed:
Adding CPU kernels for tf.scatter_nd(), tf.scatter_nd_update(), tf.scatter_nd_add(), tf.scatter_nd_sub(), tf.scatter_nd_mul() and tf.scatter_nd_div() as well as gradient functions for tf.scatter_nd() and tf.gather_nd()
Change: 138013328

sukritiramesh committed:
Update SavedModel builder to use wrapped saver's clear_devices option.
Change: 138015167

yifeif committed:
Merge remote-tracking branch 'origin/r0.11'

tensorflower-gardener committed:
Update ops-related pbtxt files.
Change: 138017353

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138017961

tensorflower-gardener committed:
Modify StochasticTensor to take an instance of a distribution.
Remove automatically generated (Distribution)Tensors.
Change: 138018661

caisq committed:
tfdbg: NodeStepper API

The Stepper API enables continue-to actions (i.e., the cont() call) on a TF graph. Stepper API and the already-existing Analyzer API constitute the base of the tfdbg Python API.

Features of the Stepper:
1) Uses TensorHandles to cache results from previous cont() calls and avoid unnecessary recomputation during subsequent cont() calls.

2) Supports cont() calls on backward optimization runs() by keeping track of which Variables are in a dirty state (updated by previous cont() calls) and restoring them if necessary. Also provides an option to disable the restoration.

3) Allows clients to override intermediate tensor values before further calls of cont().

Current limitations that will be addressed in a later CL:

1) Does not obtain intermediate tensors via the tfdbg Analyzer API. This will soon be addressed by connecting the Analyzer with the Stepper.

2) Feeding tensor handles involves an eval() call of the handle. This will be improved once TensorHandles becomes a first-class citizen in tensor types and can be fed directly.

3) Supports only a single fetch target. Support for lists, tuples and dicts of fetches will be added soon.
Change: 138019857

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138021741

benoitsteiner committed:
Register the constant op on sycl devices for numeric types only: we don't want to process strings on an accelerator.

benoitsteiner committed:
Added a device context for sycl

tensorflower-gardener committed:
Make a doc generator method public so that it can be reused in different versions of the doc generator.
Change: 138023534

Commit:
Change path functions to operate on URIs rather than paths.

- Move the URI parsing and creation functions to path.h so that we don't
  depend on core/platform from core/lib/io.
- Fixes https://github.com/tensorflow/tensorflow/issues/5316
Change: 138023841

benoitsteiner committed:
Register the sycl device under the '/device:SYCL:id' name so that its name follow the standard naming convention for devices

benoitsteiner committed:
Upgraded to the latest version of Eigen

benoitsteiner committed:
Only register the _Arg and _Retval kernel for POD types on sycl

benoitsteiner committed:
Updated the name of the sycl device

benoitsteiner committed:
Improved code formatting

tensorflower-gardener committed:
Fix capture expression of scatter_nd_op* files
Change: 138030455

Commit:
TensorBoard: Do not generate a long runs checked url by default

Right now, the names of all of the checked runs are stored in the url bar.
This means that when TensorBoard first loads, it generates a (sometimes very long) url.

With this change, TensorBoard stores the names of all the disabled runs. This has identical behavior except that the url bar will be short if all the runs are checked, and long if all the runs are unchecked.

As a general invariant, I think loading a TensorBoard without touching the UI should always generate a short url (just /scalars maybe), and the url should grow long only if you add in settings.
Change: 138032618

tensorflower-gardener committed:
Subscribe syntax for allowing side effects when tensors are executed.
Change: 138037407

benoitsteiner committed:
Fixed formatting of the BUILD file

tensorflower-gardener committed:
Delete obsolete comment.

a) EIGEN_USE_NONBLOCKING_THREAD_POOL no longer exist. The non-blocking threadpool is the default (It can be disabled by the symbol EIGEN_USE_SIMPLE_THREAD_POOL).

b) An eigen_initialize.cc file is no longer needed due to recent fixes for thread safety in Eigen.
Change: 138074328

tensorflower-gardener committed:
Add a GPU implementation of the ParameterizedTruncatedNormalOp kernel.

Benchmarks:

Benchmark                                   Time(ns)    CPU(ns) Iterations
--------------------------------------------------------------------------
BM_PTruncatedNormal_gpu_1000_1000            4632369    5175938        100  184.251M items/s
BM_PTruncatedNormal_2SD_gpu_10000_100        2849437    3368804        206  283.090M items/s
BM_PTruncatedNormal_OneTail_gpu_10000_100    3300317    3905713        179  244.174M items/s
Change: 138074670

tensorflower-gardener committed:
C++ Gradients: Adds gradient functions for some nn ops. Adds functionality to gradient checker to enable explicit initial values for some tensors.
Change: 138078095

tensorflower-gardener committed:
Replace os.walk and glob.glob with corresponding gfile methods.
Change: 138078360

Commit:
Renamed:
 * tf.all_variables to tf.global_variables
 * tf.VARIABLES -> tf.GLOBAL_VARIABLES
 * tf.initialize_all_variables -> tf.global_variable_initializers
Change: 138078998

benoitsteiner committed:
Added missing override qualifier for the FormatLibraryFileName method
Change: 138079985

malzantot committed:
Implementing WriteFile Op (#5330)

* Implementing WriteFile Op

* Addressing comments. Output value 0 means success, -1 means failure

* minor

* Changing the write_file op into a void op with no-return value

* Using OP_REQUIRES_OK(WriteStringToFile... instead of TF_CHECK_OK(...

* Code style checks using clang-format

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138083939

Commit:
after clean jenkins build some tweaks to the exclusion list

benoitsteiner committed:
Don't create SYCL kernels unless TENSORFLOW_USE_SYCL is set

philokey committed:
Fix a potential bug for android demo (#5382)

* fix bug for android demo

* fix bug for android demo

* fix bug for android demo

Commit:
Merge pull request #5365 from yifeif/master

Merge back changes from r0.11

Commit:
Expose fake quantization operators.

Add shape function for FakeQuant gradient ops.

Add a test for one of the more complicated ones.
Change: 138087659

andrewharp committed:
Allow Android to use legacy scalars for parity in model compatibility.
Change: 138089751

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138090031

gunan committed:
Create a script for building tensorflow pip package on windows using bazel.

tensorflower-gardener committed:
Fix instructions for installing cuDNN to preserve symlinks.
Change: 138090265

a7744hsc committed:
Update the misleading comment for cifar10.py's softmax_linear layer (#5259)

* Update the misunderstanding comment for cifar10.py

A fix for the issue #5251, make the comment more meaningful.

* Update comment to be a bit more precise.

* wrap to 80

leonardgithub committed:
plural should be find (#5368)

* plural should be find

plural should be find

* Update to mention matmul op output

andrewharp committed:
Refactor Android demo to put activity specific code into ClassifierActivity to simplify adding new demos. Also add some utility classes and debug rendering (tap to enable).
Change: 138091760

Commit:
Automated rollback of change 138001077
Change: 138093178

Commit:
Seal gradients interface.
Change: 138098284

tensorflower-gardener committed:
Remove duplicate PredictionKeys.
Change: 138099098

sukritiramesh committed:
Add readme for SavedModel py.
Change: 138100290

Commit:
Fix windows build.
Change: 138101249

andrewharp committed:
Fix typo in ImageUtils.getTransformationMatrix()
Change: 138103022

Commit:
Add :core_cpu headers to pip package to provide
session.h/session_options.h and some other headers that are supposed
to be publicly available.

Fixes #5380
Change: 138104302

Commit:
Merge commit for internal changes

Commit:
Gracefully fail when reading checkpoints in unsupported formats.

Also:
- Ignore checkpoints/configs that do not have any 2D tensors.
- Shield main TensorBoard from plugin failures.
Change: 138106394

Commit:
Autogenerated Change: Change TensorBoard TAG to 36
Change: 138108172

Commit:
Merge pull request #5387 from vrv/branch_138101249

Branch 138101249

Commit:
Add a .mention-bot config. (#5386)

tensorflower-gardener committed:
Add AssetManager-based FileSystem for Android.
Change: 138115819

tensorflower-gardener committed:
Making simple gradient descent work with resource variables.
Change: 138121817

Commit:
Revert "Use fast IDCT for JPEG decoding by default (#5072)"

This reverts commit 2aeedc38c604f3e9f028d696247512ea1c8522ee.

Commit:
Revert "Fix: replace deprecated nvcc flag with its replacement. (#5256)"

This reverts commit 988b53e1ba38900d1cac0fe49357d7fbeab8925a.

tensorflower-gardener committed:
Expanded docstrings for extract_dask_data and extract_dask_labels.
Change: 138123658

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138125802

andrewharp committed:
Optimize image copy loop and correct swapped R and B values.
Change: 138126747

benoitsteiner committed:
Merge remote-tracking branch 'upstream/master'

zheng-xq committed:
Add a fused adjust_hue that is 6-7 times faster on CPU. The GPU kernel will be
added in a separate CL. For now, the fused implementation can be chosen with
the environmental variable "TF_ADJUST_HUE_FUSED=1"

Before the CL:
benchmarkAdjustHue_299_299_3_cpu1 step_time: 15951.86 us
benchmarkAdjustHue_299_299_3_cpu_all step_time: 2732.68 us

After the CL:
benchmarkAdjustHue_299_299_3_cpu1 step_time: 2450.50 us
benchmarkAdjustHue_299_299_3_cpu_all step_time: 399.18 us
Change: 138128641

Commit:
Replace usages initialize_all_variables -> global_variables_initializer
Change: 138128703

Commit:
Merge pull request #5388 from vrv/master

Revert IDCT jpeg change and nvcc flag change

ebrevdo commented:
there are many changes in this code.  before a more thorough review, can you tell me if this is fully backwards compatible with the original code?  if not, how?

tensorflower-gardener committed:
Update ops-related pbtxt files.
Change: 138130752

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138131227

Commit:
Temporarily disable ASAN testing for scatter_nd.
Change: 138131457

andrewharp committed:
Update Android inference interface to take variable number of dimensions.
Change: 138133395

Commit:
Splitting Estimator interface into two:
 - Estimator that supports input_fn only (x,y,batch_size are depreacated)
 - SKCompat wrapper that takes Estimator and has x,y,batch_size arguments.

See updated tests for example usage.
Change: 138134485

tensorflower-gardener committed:
Internal cleanup.
Change: 138135876

gunan committed:
Fix: add debugging and fail-on-error to windows bazel build script.

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138136264

tensorflower-gardener committed:
Update how we force pip package to be "Root-Is-Purelib: false", as wheel 0.25.0 version
or later no long has distribution.is_pure attribute.

This change will cause the pip package tags to change based on the system it is built on. Update pip.sh to rename mac pip files(we already rename for linux) so pip names still matches our upload to gs script.
Change: 138136631

tensorflower-gardener committed:
Adds feature column functions to exported docs. Fixes #3622
Change: 138137511

langmore committed:
check_ops BUGFIX:  Call convert_to_tensor on args before doing anything else!
Change: 138138065

tensorflower-gardener committed:
Refining the condition for parallel IsDirectory(). Avoiding "defined()" for TARGET_OS_IPHONE since it is often defined as zero.  Removing "defined(__ANDROID__)" since the problem with the android emulator cannot be reproduced now.
Change: 138138237

tensorflower-gardener committed:
Switch callers of tf.pack and tf.unpack to call tf.stack and tf.unstack instead.
Change: 138139542

benoitsteiner committed:
Record which op was used for training automatically in the optimizer.
Change: 138139550

tensorflower-gardener committed:
Removed newline char in logs.
Change: 138140095

Commit:
GPUDevice: if enabling peer access fails, log a warning but do not
return an error.

On some systems and GPUs, the driver may report being able to enable
peer access between two devices, but trying to do so still fails.

The system can still run, though possibly slower than if peer access
were enabled.  Since we cannot disambiguate between supported
and unsupported cases in which this happens, we demote this
to a warning, with the exception being that if *no* device
could enable peering even though it should be possible, we
still return an error.

Fixes #5362, hopefully.
Change: 138141024

tensorflower-gardener committed:
Anonymize feature column names.
Change: 138142843

Commit:
Merge changes from github.
Change: 138143557

tensorflower-gardener committed:
Update ops-related pbtxt files.
Change: 138144503

tensorflower-gardener committed:
Update generated Python Op docs.
Change: 138144913

Commit:
tf.summary ops remove leading slashes from their names

This ensures easier conversion to the new ops, as some clients construct tag names with leading forward slashes.
Change: 138148288

caisq committed:
Minor tweak to matmul shape mismatch error message
Change: 138149179

Commit:
address review feedback

Commit:
TensorBoard compresses histograms only when the data will be stored in the reservoir.

This is a performance optimization, since histogram compression is relatively expensive.
Change: 138149635

Commit:
Loosen equality test.
Change: 138155531

Commit:
Disable ASAN for scatter_nd test (python).
Change: 138156808

phoaivu committed:
Merge branch 'phvu/fix/grid-rnn' of https://github.com/phvu/tensorflow into phvu/fix/grid-rnn

tensorflower-gardener committed:
Speed up TensorForest by limiting the leaves that FinishedNodes looks at.
Change: 138182239

tensorflower-gardener committed:
Change for internal compatibility.

Commit:
Fix performance issues with the Embedding Projector plugin.

- Setup data handlers in the class c-tor instead of on every GET request.
- Use only 1 instance of the plugin, instead of creating a new instance on every GET Request.
- Memorize readers, and other data, and share it across the multiple threads of TB server. Previously # of checkpoint readers = # of GET requests, ups.
- Checkpoints can be big (order of 1G). Reading it takes a while. Only start reading checkpoints if someone explicitly opens the embeddings tab. Previously the checkpoint was read on TB startup, which slows down TensorBoard.
Change: 138187941

caisq committed:
tfdbg: Fix a bug related to regex search and scrolling when there is line wrapping

The issue: Prior to this CL, if there are is wrapping and you do regex search and scrolling, the output pad will scroll according to line indices in the original (unwrapped) text lines instead of the wrapped text lines, which misses the actual matches. This CL fixes this issue.
Change: 138191265

sukritiramesh committed:
Add readme for SavedModel in C++.
Change: 138194799

Commit:
Partially revert OSS merge e0ed7a82756b9d38ee67cb937d73dddf42e3de6c.
Change: 138200850

Commit:
Revert to using log(1+x) instead of log1p(x).
Change: 138201323

benoitsteiner committed:
Merge pull request #5267 from benoitsteiner/master

Initial support for OpenCL

gunan committed:
Merge pull request #5385 from gunan/master

Create a script for building tensorflow pip package on windows using bazel.

Commit:
Merge commit for internal changes

Commit:
Fix config that makes file not json parseable.

Verified by loading file using python json.

llhe committed:
Add Kerberos support for HDFS file system (#5398)

* Add support for Kerberos-enabled HDFS

* Fix HDFS DeleteDir caused by HDFS-8407

llhe committed:
Remove hard coded file system scheme checking in tensorboard code (#5376)

* Remove hard coded file system scheme checking in tensorboard code

* Update URI regex pattern in tensorboard server to keep consistent with C++

guotong1988 committed:
Change Tensor summary to print tensor without losing shape information in output (#5026)

* issue 1702

* issue 1702, remove some empty line

* Update tensor.cc

* Update tensor.cc

* Update tensor.cc

* Update tensor.cc

* Update tensor.cc

* Update tensor.cc

* Update tensor.cc

* Update tensor.cc

* Update tensor_test.cc

* Update tensor.cc

* Update tensor.cc

* Update tensor_test.cc

* Update tensor.cc

* Update tensor.cc

samjabrahams committed:
Add Substr Op (#4338)

* Add initial code for substr op

* Fix naming of substr

Conflicts:
	tensorflow/core/kernels/BUILD

* Add complete substr op, tests, and documentation

* Update date 2015->2016

* Remove unnecessary header includes

* Add dtype specific tests

* Add element-wise version of Op, comments for bcast

* Removed cheeky line break in string_ops.py

* Change commented `int` vars to `size_t`

* Add case statements for static template dimensions

* Clean substr_op, update shape function, more tests

* Adds info about pos/len needing the same shape

* Remove #include of bcast statement

* Add working broadcast code with tests

* Clean line widths and comments

* Add broadcasting note to string_ops.cc

* Add print_function __future__ import to tests

* Add comment in to shape function in string_ops.cc

* Add SubtleMustCopy wrapper for scalar pos/len

* Add SubtleMustCopy to element-wise path

* Add SubtleMustCopy and FastBoundsCheck, tests

* Add examples to documentation in string_ops.cc

* Avoid repeat accessor calls to input tensor

* Reorder variable declarations in loop

* Reorder python imports to be alphabetical

Commit:
Merge pull request #5400 from vrv/branch_138201323

Branch 138201323

andreas-repp committed:
Add environment variable USE_DEFAULT_PYTHON_LIB_PATH to python_config.sh to configure usage of default python library path without user interaction. (#5397)

fabriciojoc committed:
Fixed error with NaN elements (#5024)

gunan committed:
Merge pull request #5364 from guschmue/cmaketests

enable unittests for cmake builds

girving committed:
Ignore tools/git/gen and fix a bash error in python_config.sh (#5405)

* Ignore tools/git/gen

* Avoid bash error in python_config.sh

Without this change, I get

    Please specify the location of python. [Default is /usr/bin/python]:
    Do you wish to build TensorFlow with Google Cloud Platform support? [y/N]
    No Google Cloud Platform support will be enabled for TensorFlow
    Do you wish to build TensorFlow with Hadoop File System support? [y/N]
    No Hadoop File System support will be enabled for TensorFlow
    ./util/python/python_config.sh: line 124: [: : integer expression expected
    Found possible Python library paths:
      /usr/local/lib/python2.7/dist-packages
      /usr/lib/python2.7/dist-packages
      /usr/local/buildtools/current/sitecustomize
    Please input the desired Python library path to use.  Default is [/usr/local/lib/python2.7/dist-packages]

The problem is that -eq is valid only for integers on both sides.

kbrems committed:
Update index.md (#5402)

added link for OVL

hoangmit committed:
reduce_logsumexp fix for reduction_indices.  Fixes issue #5291 (#5292)

* Fix issue 5291

* Update #5291

Update from comments

* Add unit test

* remove extra space

* Update math_ops.py

Revert back to initial approach

* wrap to 80 cols by restructuring

Commit:
Fix windows build. (#5411)

* Fix windows build.

Add resource_variable_ops to cmake

* Update the document of windows cmake build.

XuesongYang committed:
fixed the equation of the sequence increase

caisq committed:
Merge pull request #5420 from XuesongYang/patch-1

fixed the equation of the sequence increase

Ghedeon committed:
Update README.md. Wrong graph name. (#5429)

codingyourlife committed:
added full relative path to compile_ios_protobuf (#5424)

The command before and after this command have the full path. That was confusing for me and is a mistake if you ask me.

phoaivu committed:
updated grid_rnn_test

phoaivu committed:
resolved conflict

phoaivu committed:
Merge branch 'phvu/fix/grid-rnn' of https://github.com/phvu/tensorflow into phvu/fix/grid-rnn

phvu commented:
Sorry I screwed up this PR (lack of water and sleep apparently). Will close this and submit another PR, with responds to your comments.
